Source/DafnyLanguageServer/Handlers/DafnyWorkspaceSymbolHandler.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Dafny.LanguageServer.Workspace;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Microsoft.Dafny.LanguageServer.Handlers {
+  public class DafnyWorkspaceSymbolHandler : WorkspaceSymbolsHandlerBase {
+    private readonly ILogger logger;
+    private readonly IProjectDatabase projects;
+    private readonly DafnyOptions dafnyOptions;
+
+    public DafnyWorkspaceSymbolHandler(ILogger<DafnyWorkspaceSymbolHandler> logger, IProjectDatabase projects, DafnyOptions dafnyOptions) {
+      this.logger = logger;
+      this.projects = projects;
+      this.dafnyOptions = dafnyOptions;
+    }
+
+    protected override WorkspaceSymbolRegistrationOptions CreateRegistrationOptions(WorkspaceSymbolCapability capability,
+      ClientCapabilities clientCapabilities) {
+      return new WorkspaceSymbolRegistrationOptions {
+        WorkDoneProgress = false
+      };
+    }
+
+    public override async Task<Container<SymbolInformation>?> Handle(WorkspaceSymbolParams request, CancellationToken cancellationToken) {
+      var queryText = request.Query.ToLower();
+      Dictionary<SymbolInformation, int> result = new Dictionary<SymbolInformation, int>();
```

Comment:
Consider using `var` on the lefthand. What editor are you using to write C# ? Rider will give you suggestions to apply such refactorings.

Reply:
Sounds good, I'll change that. I'm also using Rider, but must have missed that suggestion.